idx/memory/memory_test.go

```diff
@@ -40,29 +40,51 @@ func getRandomString(n int, alphabets ...byte) string {
 	return string(bytes)
 }
 
-func getMetricData(orgId, depth, count, interval int, prefix string) []*schema.MetricData {
+func getMetricData(orgId, depth, count, interval int, prefix string, tagged bool) []*schema.MetricData {
 	data := make([]*schema.MetricData, count)
 	series := getSeriesNames(depth, count, prefix)
-	for i, s := range series {
 
+	for i, s := range series {
 		data[i] = &schema.MetricData{
 			Name:     s,
 			Metric:   s,
 			OrgId:    orgId,
 			Interval: interval,
 		}
+		if tagged {
+			data[i].Tags = []string{fmt.Sprintf("series_id=%d", i)}
+		}
 		data[i].SetId()
 	}
+
 	return data
 }
 
+// testWithAndWithoutTagSupport calls a test with all combinations of
+// the settings TagSupport and tagsInTree. In some cases those settings can
+// affect the logic quite a lot, so we need to test all combinations
+func testWithAndWithoutTagSupport(t *testing.T, f func(*testing.T)) {
```

Comment:
this is a cool idea, but the mentioned tagsInTree parameter does not exist

Reply:
wouldn't it be more meaningful though, if the called functions actually did more with tags?
e.g. `testWithAndWithoutTagSupport(t, testPrune)`  testPrune doesn't have any tagged data, so the tagsupport enabling won't do much.  so the tag support aspect of this is a bit misleading. or should we just document that the purpose here is testing solely non-tag-related stuff in context of an enabled and disabled index?

(maybe out of scope for this PR?)

Reply:
sorry, that comment is old, the `tagsInTree` parameter did exist but then we decided to remove it.

Reply:
I think even if a function shouldn't do anything related to tagged series, it's still a good idea to test it with both settings, since doing so is basically free and it just gives a little more assurance that everything works as expected.

Reply:
https://github.com/grafana/metrictank/pull/806/commits/ad7491ae4bcc2aa656882397f3f93ef484717e0e